pkg/datastore/pagination/iterator.go

```diff
@@ -0,0 +1,114 @@
+package pagination
+
+import (
+	"context"
+	"errors"
+
+	"github.com/authzed/spicedb/internal/datastore/common"
+	"github.com/authzed/spicedb/pkg/datastore"
+	"github.com/authzed/spicedb/pkg/datastore/options"
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+)
+
+// NewPaginatedIterator creates an implementation of the datastore.Iterator
+// interface that internally paginates over datastore results.
+func NewPaginatedIterator(
```

Comment:
Does not have a test, but perhaps as I proposed in `pkg/datastore/test/pagination.go`, we could do integration test for the iterator there